lib/exporter.rb

```diff
@@ -0,0 +1,370 @@
+    delete_temp_files(path)
+
+    [x1,y1]
+  end
+
+  ########################
+  ## Run on maps:
+
+  # distort all warpables, returns upper left corner coords in x,y
+  def self.distort_warpables(scale, warpables, export, slug)
+
+    puts '> generating geotiffs of each warpable in GDAL'
+    lowest_x=0
+    lowest_y=0
+    warpable_coords = []
+    current = 0
+    warpables.each do |warpable|
+     current += 1
+
+     ## TODO: refactor to generate static status file:
+     export.status = 'warping '+current.to_s+' of '+warpables.length.to_s
+     puts 'warping '+current.to_s+' of '+warpables.length.to_s
+     export.save
+     ## 
+
+     my_warpable_coords = warpable.generate_perspectival_distort(scale,slug)
+     puts '- '+my_warpable_coords.to_s
+     warpable_coords << my_warpable_coords
+     lowest_x = my_warpable_coords.first if (my_warpable_coords.first < lowest_x || lowest_x == 0)
+     lowest_y = my_warpable_coords.last if (my_warpable_coords.last < lowest_y || lowest_y == 0)
+    end
+    [lowest_x,lowest_y,warpable_coords]
+  end
+
+  # generate a tiff from all warpable images in this set
+  def self.generate_composite_tiff(coords, origin, placed_warpables, slug, ordered)
+    directory = "public/warps/"+slug+"/"
+    composite_location = directory+slug+'-geo.tif'
+    geotiffs = ''
+    minlat = nil
+    minlon = nil
+    maxlat = nil
+    maxlon = nil
+    placed_warpables.each do |warpable|
+      warpable.nodes_array.each do |n|
+        minlat = n.lat if minlat == nil || n.lat < minlat
+        minlon = n.lon if minlon == nil || n.lon < minlon
+        maxlat = n.lat if maxlat == nil || n.lat > maxlat
+        maxlon = n.lon if maxlon == nil || n.lon > maxlon
+      end
+    end
+    first = true
+    if ordered != true
+      # sort by area; this would be overridden by a provided order
+      warpables = placed_warpables.sort{|a,b|b.poly_area <=> a.poly_area}
+    end
+    warpables.each do |warpable|
+      geotiffs += ' '+directory+warpable.id.to_s+'-geo.tif'
+      if first
+        gdalwarp = "gdalwarp -s_srs EPSG:3857 -te "+minlon.to_s+" "+minlat.to_s+" "+maxlon.to_s+" "+maxlat.to_s+" "+directory+warpable.id.to_s+'-geo.tif '+directory+slug+'-geo.tif'
+        first = false
+      else
+        gdalwarp = "gdalwarp "+directory+warpable.id.to_s+'-geo.tif '+directory+slug+'-geo.tif'
+      end
+      puts gdalwarp
+      system(self.ulimit+gdalwarp)
+    end
+    composite_location
+  end
+
+  # generates a tileset at root/public/tms/<slug>/
+  # root is something like https://mapknitter.org
+  def self.generate_tiles(key, slug, root)
+    key = "AIzaSyAOLUQngEmJv0_zcG1xkGq-CXIPpLQY8iQ" if key == "" # ugh, let's clean this up!
+    key = key || "AIzaSyAOLUQngEmJv0_zcG1xkGq-CXIPpLQY8iQ"
+    gdal2tiles = 'gdal2tiles.py -k --s_srs EPSG:3857 -t "'+slug+'" -g "'+key+'" '+root+'/public/warps/'+slug+'/'+slug+'-geo.tif '+root+'/public/tms/'+slug+"/"
```

Comment:
Hi @icarito if you have a moment, i'm a bit stuck here; i think we're down to the GDAL version on this last command. I've been trying to run it locally. 

Here are the files that this should be runnable on: https://publiclab.org/i/29584.zip

Reply:
`gdal2tiles.py -k --s_srs EPSG:3857 -t "saugus-landfill-incinerator" -g "AIzaSyAOLUQngEmJv0_zcG1xkGq-CXIPpLQY8iQ" /app/public/warps/saugus-landfill-incinerator/saugus-landfill-incinerator-geo.tif /app/public/tms/saugus-landfill-incinerator/`

Reply:
And Mapknitter.org uses: 
```
$ gdal2tiles.py --version
GDAL 2.1.2, released 2016/10/24
```

Reply:
Travis was same: `GDAL 2.1.2, released 2016/10/24`